roboplot/core/plotter.py

```diff
@@ -0,0 +1,76 @@
+import roboplot.core.curves as curves
+import roboplot.core.debug_movement as debug_movement
+import roboplot.core.liftable_pen as liftable_pen
+import roboplot.core.stepper_control as stepper_control
+
+
+class Plotter:
+    default_pen_speed = 100  # I.e. as fast as possible
+    default_resolution = 0.1
+
+    def __init__(self, axes: stepper_control.AxisPair, pen: liftable_pen.LiftablePen):
+        self._axes = axes
+        self._pen = pen
+
+    def home(self):
+        self._pen.lift()
+        self._axes.home()
+
+    def draw(self, curve_list, pen_speed: float = default_pen_speed, resolution:float=default_resolution):
+        """Draw the curve (with the pen down), lifting the pen before and after."""
+
+        if isinstance(curve_list, curves.Curve):
+            curve_list = [curve_list]
+
+        self._lift_pen()
+        if len(curve_list) > 0:
+            self._move_to_start_of_curve(curve_list[0], pen_speed, resolution)
+            self._drop_pen()
+            for curve in curve_list:
+                self._axes.follow(curve, pen_speed, resolution)
+            self._lift_pen()
+
+    def follow(self, curve_list, pen_speed: float = default_pen_speed, resolution:float=default_resolution):
+        """Follow a curve, with the pen up."""
+        # TODO: It is not ideal to not be changing colour each curve when the pen is up...
+        if isinstance(curve_list, curves.Curve):
+            curve_list = [curve_list]
+
+        self._lift_pen()
+        for curve in curve_list:
+            self._axes.follow(curve, pen_speed, resolution)
+
+    def _lift_pen(self):
+        self._pen.lift()
+
+    def _drop_pen(self):
+        self._pen.drop()
+
+    def _move_to_start_of_curve(self, curve, pen_speed, resolution):
+        self._axes.follow(
```

Comment:
So I'm going to treat this like the house of commons and the house of lords and send this back to you again, though you're still free to stick with your guns!!

My reasoning behind doing it that way is:
  - It is a clean way to ensure a consistent state after we call `draw(...)`. That is, if `_move_to_start_of_curve(...)` contained the pen lift, then I would still need to end `draw(...)` with a `self._lift_pen()` to ensure that the pen is always lifted at the end of the `draw(...)` method.
  - Currently the `_move_to_start_of_curve(...)` method does exactly what it says on the tin and there is (hopefully!) no confusion. If we move `self._pen_lift()` inside this then we would either have to rename it to `_lift_pen_and_move_to_start_of_curve(...)` (which is a bit of a mouthful) or live with implicit behaviour which isn't documented in the method title.

To [quote](http://alvinalexander.com/programming/clean-code-quotes-robert-c-martin) Robert Martin ("uncle bob") in *Clean Code* (pp. 39):

> "You know you are working on clean code when each routine you read turns out to be pretty much what you expected."

Also (pp. 35)

> "Functions should do one thing. They should do it well. The should do it only."